### PR TITLE
Did you mean typo corrections?

### PR DESCRIPTION
**Pull request by EvilSourcerer**
this
image
and this
image

### CODE REVIEW

**Comment by usereewrjrjeejr** [comment]
Cringe Normie

**Comment by babbaj** [comment]
>unsigned commits

major cringe
close this please

**Comment by EvilSourcerer** [comment]
signed 🕋

**Comment by babbaj** [comment]
still not signed idiot

**Comment by usereewrjrjeejr** [comment]
If you want it signed do it yourself

**Comment by EvilSourcerer** [comment]
that's not how it works

**Comment by usereewrjrjeejr** [comment]
I don't give 2 shits

**Comment by EvilSourcerer** [comment]
YES! it's verified finally

**Comment by usereewrjrjeejr** [comment]
@babbaj

**Comment by usereewrjrjeejr** [comment]
@babbaj

**Comment by leijurv** [comment]
unknown (2)

**Comment by leijurv** [comment]
304ond

**Comment by usereewrjrjeejr** [comment]
@babbaj

File: src/api/java/baritone/api/utils/ExampleBaritoneControl.java [hunk 1]
```diff
@@ -90,8 +91,35 @@ public void onSendChatMessage(ChatEvent event) {
         String msg = event.getMessage();
         if (BaritoneAPI.getSettings().prefixControl.value && msg.startsWith(COMMAND_PREFIX)) {
             if (!runCommand(msg.substring(COMMAND_PREFIX.length()))) {
-                logDirect("Invalid command");
+                String closestmatch = "";
+                int lowestnum = -1;
+                for (Field field : BaritoneAPI.getSettings().getClass().getFields()) {
```
[the remark this answers is not in the document]
Comment (0-x-2-2) [reply src/api/java/baritone/api/utils/ExampleBaritoneControl.java]: shit version of whats already in the settings class yikes! reinvesting but worse 😎

File: src/api/java/baritone/api/utils/ExampleBaritoneControl.java [hunk 2]
```diff
@@ -90,8 +91,35 @@ public void onSendChatMessage(ChatEvent event) {
...
+                    if (Levenshtein.distance(command, msg.split(" ")[0]) < lowestnum && lowestnum != -1) {
+                        closestmatch = command;
+                        lowestnum = Levenshtein.distance(command, msg.split(" ")[0]);
+                    }
+                    if (lowestnum == -1) {
+                        closestmatch = command;
+                        lowestnum = Levenshtein.distance(command, msg.split(" ")[0]);
+                    }
+                }
+                logDirect("You fool. You absolute buffoon. That command doesn't exist. Did you mean " + closestmatch + "?");
```
Comment (0-x-2-2) [inline src/api/java/baritone/api/utils/ExampleBaritoneControl.java]: bigotry is bad!

File: src/api/java/baritone/api/utils/ExampleBaritoneControl.java [hunk 3]
```diff
@@ -90,8 +91,35 @@ public void onSendChatMessage(ChatEvent event) {
...
+                            closestmatch = fieldname;
+                        }
+                        if (lowestnum == -1) {
+                            closestmatch = fieldname;
+                            lowestnum = Levenshtein.distance(fieldname, msg.split(" ")[0]);
+                        }
+                    }
+                }
+                for (int i = 0; i < HELP_MSG.split("\n").length; i++) {
+                    String command = HELP_MSG.split("\n")[i].split(" ")[0];
```
Comment (0-x-2-2) [inline src/api/java/baritone/api/utils/ExampleBaritoneControl.java]: 🤡 
major performance issue

File: src/api/java/baritone/api/utils/Levenshtein.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Baritone.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package baritone.api.utils;
+
+public final class Levenshtein {
```
Comment (0-x-2-2) [inline src/api/java/baritone/api/utils/Levenshtein.java]: 🤡

File: src/api/java/baritone/api/utils/Levenshtein.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    public static int distance(String stringa, String stringb) {
```
Comment (0-x-2-2) [inline src/api/java/baritone/api/utils/Levenshtein.java]: 🦍

File: src/api/java/baritone/api/utils/Levenshtein.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+        String tempa = stringa.toLowerCase();
+        String tempb = stringb.toLowerCase();
+        // i == 0
+        int[] costs = new int[stringb.length() + 1];
+        for (int j = 0; j < costs.length; j++)
```
Comment (0-x-2-2) [inline src/api/java/baritone/api/utils/Levenshtein.java]: OY VEY!

Comment (0-x-2-2) [reply src/api/java/baritone/api/utils/Levenshtein.java]: leijurv wont like this line

File: src/api/java/baritone/api/utils/ExampleBaritoneControl.java [hunk 7]
```diff
@@ -90,8 +91,35 @@ public void onSendChatMessage(ChatEvent event) {
         String msg = event.getMessage();
         if (BaritoneAPI.getSettings().prefixControl.value && msg.startsWith(COMMAND_PREFIX)) {
             if (!runCommand(msg.substring(COMMAND_PREFIX.length()))) {
-                logDirect("Invalid command");
+                String closestmatch = "";
+                int lowestnum = -1;
+                for (Field field : BaritoneAPI.getSettings().getClass().getFields()) {
+                    if (field.getType().isAssignableFrom(Settings.Setting.class)) {
+                        String fieldname = field.getName();
+                        if (Levenshtein.distance(fieldname, msg.split(" ")[0]) < lowestnum && lowestnum != -1) {
```
Comment (0-x-2-2) [inline src/api/java/baritone/api/utils/ExampleBaritoneControl.java]: baritone_2019_picturized

File: src/api/java/baritone/api/utils/ExampleBaritoneControl.java [hunk 8]
```diff
@@ -90,8 +91,35 @@ public void onSendChatMessage(ChatEvent event) {
         String msg = event.getMessage();
         if (BaritoneAPI.getSettings().prefixControl.value && msg.startsWith(COMMAND_PREFIX)) {
             if (!runCommand(msg.substring(COMMAND_PREFIX.length()))) {
-                logDirect("Invalid command");
```
Comment (0-x-2-2) [inline src/api/java/baritone/api/utils/ExampleBaritoneControl.java]: based and redpilled code

File: src/api/java/baritone/api/utils/ExampleBaritoneControl.java [hunk 9]
```diff
@@ -90,8 +91,35 @@ public void onSendChatMessage(ChatEvent event) {
...
+                String closestmatch = "";
```
Comment (0-x-2-2) [inline src/api/java/baritone/api/utils/ExampleBaritoneControl.java]: 🤢 code

**Comment by 0-x-2-2** [comment]
this code is communism. I recommend you close this pull request. @leijurv @ZeroMemes 

Or else...
baritone_2019_colorized

**Comment by leijurv** [comment]
OwO communism?!

**Comment by 0-x-2-2** [comment]
make america fight communism again!

File: src/api/java/baritone/api/utils/Levenshtein.java [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+            costs[j] = j;
+        for (int i = 1; i <= tempa.length(); i++) {
+            // j == 0; nw = lev(i - 1, j)
+            costs[0] = i;
+            int nw = i - 1;
+            for (int j = 1; j <= stringb.length(); j++) {
+                int cj = Math.min(1 + Math.min(costs[j], costs[j - 1]), tempb.charAt(i - 1) == stringb.charAt(j - 1) ? nw : nw + 1);
```
Comment (leijurv) [inline src/api/java/baritone/api/utils/Levenshtein.java]: smells like pasta

Comment (leijurv) [reply src/api/java/baritone/api/utils/Levenshtein.java]: license violation?

**Comment by usereewrjrjeejr** [comment]
When

**Pull request closed without merging**